src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -664,121 +811,311 @@ TEST_CASE(
     "correctly")
 {
   PerfAnalyzerParameters params{};
+  params.user_data = {"fake_file.json"};
   bool is_sequence_model{false};
-  const auto& ParameterizeAsyncAndStreaming{[](bool& async, bool& streaming) {
-    SUBCASE("sync non-streaming")
+
+  std::vector<std::vector<int32_t>> expected_results;
+  std::vector<ModelTensor> tensors;
+  bool expect_init_failure = false;
+  bool expect_thread_failure = false;
+
+  ModelTensor model_tensor1{};
+  model_tensor1.datatype_ = "INT32";
+  model_tensor1.is_optional_ = false;
+  model_tensor1.is_shape_tensor_ = false;
+  model_tensor1.name_ = "INPUT1";
+  model_tensor1.shape_ = {1};
+
+  ModelTensor model_tensor2 = model_tensor1;
+  model_tensor2.name_ = "INPUT2";
+
+  std::string json_str{R"({
+   "data": [
+     { "INPUT1": [1], "INPUT2": [21] },
+     { "INPUT1": [2], "INPUT2": [22] },
+     { "INPUT1": [3], "INPUT2": [23] }     
+   ]})"};
```

Comment:
Is having multiple streams not supported for non-sequence models? Do we need to have similar testing for sequence models too? It might already exist. I just couldn't find it.

Reply:
Good question. I think that multiple streams is not supported for non-sequence, and is supported for sequence.  I'll file a follow-on ticket to investigate, confirm, and protect via testing.